src/image/pixels.js

```diff
@@ -614,7 +614,8 @@ p5.prototype.filter = function(...args) {
       // dest coords
       -this.width/2, -this.height/2, this.width, this.height
     );
-
+    //clearing the main canvas
+    this._renderer.clear();
     // filter it with shaders
     this.filterGraphicsLayer.filter(operation, value);
```

Comment:
I think we might still need to put the `image()` call below in between a `push()`, `resetMatrix()`, and `pop()` similar to what we do for WebGL, since it looks like it still has an offset here:

https://editor.p5js.org/davepagurek/sketches/PowzoU-_P

Reply:
Okay.